cli/cli/infra.py

```diff
@@ -159,6 +161,26 @@ def _get_namespace(self):
 
         return None
 
+    def get_function_endpoint(self, namespace_name, function_name):
+        namespaces = self.functions.list_namespaces(
+            region=API_REGION,
+            name=namespace_name,
+        )
+        if len(namespaces.namespaces) == 0:
+            click.secho("No namespace found", fg="red", bold=True)
+            raise click.Abort()
+
+        namespace = namespaces.namespaces[0]
+        functions: ListFunctionsResponse = self.functions.list_functions(
+            namespace_id=namespace.id, region=API_REGION
+        )
+
+        for f in functions.functions:
+            if f.name == function_name:
+                return f.domain_name
+
+        return None
```

Comment:
(If you use the name filter from the API:)

```suggestion
   return functions[0] if functions else None
```